src/sentry/runner/commands/devserver.py

```diff
@@ -78,6 +78,15 @@ def devserver(reload, watchers, workers, bind):
         parsed_url = urlparse(url_prefix)
         https_port = str(parsed_url.port or 443)
         https_host = parsed_url.hostname
+
+        # Determine a random port for the backend http server
```

Comment:
This is possible to be a race condition if something claims the port between the time we check and the time we actually try to bind to it, but I think we can live with this.